src/js/app.js

```diff
@@ -0,0 +1,196 @@
+          reader.onload = function (event) {
+            try {
+              const result = JSON.parse(event.target.result);
+              scope.$apply(() => {
+                scope.uploadedFiles[file.name] = result;
+              });
+            } catch (e) {
+              scope.$apply(() => {
+                alerts("invalid json. pls refer to index guide ",
+                  "danger", true, 5000);
+              });
+              return;
+            }
+          };
+          reader.readAsText(file);
+        });
+      });
+    }
+    return {
+      template: template,
+      link: link
+    };
+  }])
+  .controller('mainController', ['$scope', 'alerts', ($scope, alerts) => {
+    /**
+     *  {Object} instanciates the invertedIndex class
+     */
+    const indexer = new InvertedIndex();
+
+    /**
+     *  {Object}  uploaded files
+     */
+    $scope.uploadedFiles = {};
+
+    /** 
+     *  {Object} indexed value to display
+     */
+    $scope.index = [];
+
+    /** 
+     *  {Boolean} shwows indexed table
+     */
+    $scope.showTable = false;
+
+    /** 
+     *  {Array} Holds number of data in a file
+     */
+    $scope.fileCount = [];
+
+    /** 
+     *  {Array} Holds all indexed files
+     */
+    $scope.allIndexedFiles = [];
+
+
+    /** set the value to be selected
+     * @param {String} dom element to get
+     * @return {String} selected value
+     */
+    $scope.selected = (id) => {
+      return document.getElementById(id || 'indexed-files').value;
+    };
+
+
+    /**
+     * @param  {String} fileName - file name
+     * @param  {Object} data  - data to transform
+     * @param  {count} count - count of object in the file
+     */
+    $scope.transformData = (fileName, data, count) => {
+      return {
+        fileName: fileName,
+        data: data,
+        count: count
+      };
+    };
+
+    /**
+     * creates an inverted Index 
+     */
+    $scope.createIndex = () => {
+      const selected = $scope.selected('uploaded-files');
+      const uploadedFiles = $scope.uploadedFiles;
+      console.log(selected);
+      if ($scope.isValidFile(selected) &&
+        uploadedFiles.hasOwnProperty(selected)) {
+        const data = indexer.createIndex(selected, uploadedFiles[selected]);
+        if (data === 'error' || $scope.length(data) < 1) {
+          alerts('invalid json file', 'danger');
+          return false;
+        }
+        const fileCount = uploadedFiles[selected].length;
+        $scope.index[0] = $scope.transformData(selected, data, fileCount);
+        $scope.fileCount[selected] = fileCount;
+        $scope.allIndexedFiles.push(selected);
+        $scope.showTable = true;
+      }
+    };
+
+    /** searches for a query
+     *
+     * @param {String} query = value to be searched
+     * @return {void}  set the index value
+     */
+    $scope.searchIndex = (query) => {
+      let result;
+      const selected = document.getElementById('indexed-files').value;
```

Comment:
ng-model was given me issues. I took the easy way out. I will look  into fixing that